lib/inspec/rspec_json_formatter.rb

```diff
@@ -465,106 +582,160 @@ def format_lines(lines, indentation)
     lines.gsub(/\n/, "\n" + indentation)
   end
 
-  # Sorts through results, calls print_line (called from flush_current_control)
-  def print_results(all)
-    all.each do |x|
-      test_status = x[:status_type]
-      test_color = COLORS[test_status]
-      indicator = INDICATORS[x[:status]]
-      indicator = INDICATORS['empty'] if indicator.nil?
-      if x[:message]
-        msg = x[:code_desc] + "\n" + x[:message]
+  #
+  # This class wraps a control hash object to provide a useful inteface for
+  # maintaining the associated profile, ids, results, title, etc.
+  #
+  class Control
+    include Comparable
+
+    def <=>(other)
```

Comment:
Comparable is implemented here not for sorting but for better comparison between controls to ensure that similar named controls in different profiles are considered separate things.